snow/networking/handler/handler.go

```diff
@@ -709,7 +709,20 @@ func (h *handler) handleSyncMsg(ctx context.Context, msg Message) error {
 		}
 
 		return engine.Chits(ctx, nodeID, msg.RequestId, preferredID, preferredIDAtHeight, acceptedID, msg.AcceptedHeight)
+	case *p2ppb.Simplex:
+		if engineType != p2ppb.EngineType_ENGINE_TYPE_SIMPLEX {
+			h.ctx.Log.Debug("dropping simplex message",
+				zap.String("reason", "engine type mismatch"),
+				zap.String("messageOp", op),
+				zap.Stringer("currentEngineType", currentState.Type),
+			)
+			return nil
+		}
 
+		return fmt.Errorf(
```

Comment:
I am not sure we want to return an error here, this will stop the chain, won't it?